crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -28,18 +28,53 @@ use crate::Db;
 pub(crate) struct AstIds {
     /// Maps expressions to their expression id. Uses `NodeKey` because it avoids cloning [`Parsed`].
     expressions_map: FxHashMap<ExpressionNodeKey, ScopedExpressionId>,
+    uses_map: FxHashMap<ExpressionNodeKey, ScopedUseId>,
```

Comment:
Would it make sense to instead have an `IndexVec` from `ScopedExpressionId` to `ScopedUseId` or are we using a map because only very few expression ids have `ScoipedUseId`?

Reply:
Yes, I did it this way because it's quite sparse; basically only `Name` expressions are uses.